models/reviewDao.js

```diff
@@ -0,0 +1,161 @@
+const { appDataSource } = require("./dbconfig");
+
+const getReviewData = async (userId, shopId) => {
+  const countRecommendByUserQuery = function queryBuild(userId) {
+    let clause = "";
+    if (userId) {
+      clause = clause +
+        `(
+        SELECT count(*)
+        FROM review_recommends
+        WHERE 
+          review_id = r.id
+        AND
+          user_id = ${userId}
+      ) as recommendByThisUser,`;
+    }
+    return clause;
+  };
+
+  return appDataSource.query(
+    `SELECT
+	    r.id as reviewId,
+      u.nickname,
+      r.score as reviewScore,
+      r.content as reviewContent,
+      ${countRecommendByUserQuery(userId)}
+      (
+        SELECT count(*) 
+        FROM review_recommends
+        WHERE review_id = r.id
+      ) as recommendCount
+    FROM
+      reviews as r
+    JOIN users as u ON u.id = r.user_id
+    WHERE
+      r.shop_id = ?;
+    `,
+    [shopId]
+  );
+};
+
+const createReviewData = async (content, score, userId, shopId) => {
+  const queryRunner = appDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+
+  try {
+    await queryRunner.query(
+      `INSERT INTO reviews (
+        content,
+        score,
+        user_id,
+        shop_id
+      ) VALUES (?, ?, ?, ?)
+      `,
+      [content, score, userId, shopId]
+    );
+
+    // 리뷰 평균 구해서 shops의 average_rating에 update.
+    const ReviewScoresArr = await queryRunner.query(
```

Comment:
```suggestion
    const reviewScores = await queryRunner.query(
```
컨벤션 수정입니다. 대소문자 + Scores 자체가 복수단위의 단어이므로 Arr은 지워주셔도 됩니다.